src/functions/src/submitContactForm.ts

```diff
@@ -1,10 +1,46 @@
 import axios from 'axios';
 import * as admin from 'firebase-admin';
 import * as functions from 'firebase-functions';
+import { ApolloClient, createHttpLink, gql, InMemoryCache } from '@apollo/client';
+import { setContext } from '@apollo/client/link/context';
+// import { ApolloClient, createHttpLink, InMemoryCache, gql } from "@apollo/client";
+// import { setContext } from '@apollo/client/link/context';
 
 if (admin.apps.length === 0) {
   admin.initializeApp();
 }
+export async function createDiscussion(topic: string, body: string) {
+  const httpLink = createHttpLink({
+    uri: 'https://api.github.com/graphql',
+  });
+
+  const authLink = setContext((_, { headers }) => {
+    return {
+      headers: {
+        ...headers,
+        authorization: `Bearer ${process.env.GITHUB_ACCESS_TOKEN}`,
+      }
+    }
+  });
+
+  const client = new ApolloClient({
+    link: authLink.concat(httpLink),
+    cache: new InMemoryCache()
+  });
+
+  
+  const { data } = await client.mutate({
+    mutation: gql`
+    {
+       # input type: CreateDiscussionInput
+        createDiscussion(input: {repositoryId: "MDEwOlJlcG9zaXRvcnkyNjg2NzA0NjY=", categoryId: "DIC_kwDOEAOWAs4COX3q", body: "${body}", title: "${topic}"}) {
```

Comment:
how do you get the repository id/category id? i'm trying to test it on a test repo first

Reply:
https://docs.github.com/en/graphql/overview/explorer I used this